WeatherApp_Obj-C/WeatherApp_Obj-C/Screens/SignInScreen/SignInViewController.m

```diff
@@ -0,0 +1,67 @@
+//
+//  SignInViewController.m
+//  WeatherApp_Obj-C
+//
+//  Created by Muradasil Birhan on 14/02/2020.
+//  Copyright © 2020 RodeApps. All rights reserved.
+//
+
+#import "SignInViewController.h"
+#import "UIColor+CustomColors.h"
+
+@interface SignInViewController ()
+
+@end
+
+@implementation SignInViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.signInButton.backgroundColor = [UIColor greenButton];
+    self.signInButton.layer.cornerRadius = 25;
+}
+
+- (void)viewWillAppear:(BOOL)animated
+{
+    [super viewWillAppear:animated];
+    [self.navigationController setNavigationBarHidden:NO animated:YES];
+    [self.navigationController.navigationBar setBarTintColor:[UIColor whiteColor]];
+    [self.navigationController.navigationBar setShadowImage:[[UIImage alloc] init]];
+}
+- (IBAction)back:(id)sender {
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (IBAction)signInButtonClicked:(id)sender {
+    [self resetErrors];
+    if ([self validateFields]) {
+        [self showLoading];
+        [[UserRepository new] loginUser:self.emailInput.userInput.text withPassword:self.passwordInput.userInput.text success:^{
+            [self hideLoading];
+            [self performSegueWithIdentifier:@"GoToHomeScreen" sender:nil];
+        } error:^(NSString * _Nullable error) {
+            [self hideLoading];
+            NSLog(@"%@", error);
+            [self showError:error];
+        }];
+    }
+}
+
+- (BOOL)validateFields {
+    BOOL isValid = YES;
+    if ([self.emailInput.userInput.text isEqual:@""]) {
```

Comment:
Rather than checking for isEqual or isEqualToString, check if length is equal to 0.
Some versions of iOS return nil for textfields with no input